conans/client/cmd/export_pkg.py

```diff
@@ -54,13 +54,22 @@ def _init_conanfile_infos():
     recipe_hash = layout.recipe_manifest().summary_hash
     conanfile.info.recipe_hash = recipe_hash
     conanfile.develop = True
-    conanfile.folders.set_base_build(build_folder)
-    conanfile.folders.set_base_source(source_folder)
-    conanfile.folders.set_base_package(dest_package_folder)
-    conanfile.folders.set_base_install(install_folder)
+    if hasattr(conanfile, "layout"):
+        conanfile_folder = os.path.dirname(source_conanfile_path)
+        conanfile.folders.set_base_build(conanfile_folder)
+        conanfile.folders.set_base_source(conanfile_folder)
+        conanfile.folders.set_base_package(dest_package_folder)
+        conanfile.folders.set_base_install(conanfile_folder)
+        conanfile.folders.set_base_generators(conanfile_folder)
```

Comment:
Similar to https://github.com/conan-io/conan/pull/9668:

- I need to know the location of the generators_folder.
- I prefer not needing an extra ``--build-folder`` cli argument